docs/src/test/java/docs/http/javadsl/Http2Test.java

```diff
@@ -32,5 +36,12 @@ void testBindAndHandleAsync() {
         toHostHttps("127.0.0.1", 8443).withCustomHttpsContext(httpsConnectionContext),
         materializer);
     //#bindAndHandleAsync
+
+    //#bindAndHandleRaw
+
+    // TODO is there a reason there is no `bindAndHandle` that accepts
+    // a `ConnectionContext` in the javadsl?
```

Comment:
Is there? Should we add it?

Reply:
In Java connection context is carried in ConnectHttp, the DSL that kept our sanity since otherwise too many overloads were needed:

```java
  def bindAndHandle(
    handler:      Flow[HttpRequest, HttpResponse, _],
    connect:      ConnectHttp,
```


```java
abstract class ConnectHttp {
  def host: String
  def port: Int

  def isHttps: Boolean
  def connectionContext: Optional[HttpsConnectionContext]
```